### PR TITLE
Build page data deux

### LINKED ISSUE #151: Build component loads data from API
Follow up to #120.

AC
- When clicking a build on the project page, the ID in the HREF is used to load build data from the brigade API.

### PR DESCRIPTION
**Pull request by seanknox**
Note: merge https://github.com/Azure/kashti/pull/158 first before this one. This is branched off of it.

This builds upon @flynnduism's work in #152. Due to all the change in mock data and it was easier to re-apply his changes on top of a fresh branch.

Closes #151

### CODE REVIEW

File: src/app/services/project/api-project.service.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export class ApiProjectService implements ProjectService {
       .get<ProjectsBuild[]>(projectBuildsUrl, httpOptions);
   }
 
-  getProject(projectId): Observable<Project> {
+  getProject(projectId: string): Observable<Project> {
```
Comment (radu-matei) [inline src/app/services/project/api-project.service.ts]: 👍

File: src/app/build/build.component.html [hunk 2]
```diff
@@ -6,35 +6,49 @@
...
           </h1>
 
           <div class="build-status success" ng-class="{
             'success': build.worker.status == 'Succeeded',
             'failed': build.worker.status == 'Failed',
-            'active': build.worker.status == 'Pending'}" style="">
-            <div class="success" ng-show="build.worker.status == 'Succeeded'" style=""><i class="icon ion-md-checkmark-circle"></i> Passed</div>
+            'active': build.worker.status == 'Pending'}">
+            <div class="success" ng-show="build.worker.status == 'Succeeded'">
+              <i class="icon ion-md-checkmark-circle"></i> {{ build?.status }}</div>
```
Comment (radu-matei) [inline src/app/build/build.component.html]: Nice!

Comment (radu-matei) [inline src/app/build/build.component.html]: 👍

File: src/app/build/build.component.html [hunk 3]
```diff
@@ -59,17 +74,20 @@ <h4>
...
           )" class="ng-binding"> brigade-cli</em>  -->
-          : <em class="ng-binding">exec</em> 
-          started a build via <em class="ng-binding">commit #master</em>
+        :
+        <em class="ng-binding">{{ build?.type }}</em>
+        started a build via
+        <em class="ng-binding">commit {{ revision?.commit }}</em>
       </li>
 
       <!-- Pending API hookup:
```
Comment (radu-matei) [inline src/app/build/build.component.html]: This is hard-coded so far, right? 
We need to update the services (both the mock and the real ones) to get this data - do we have all Brigade API methods in place to plug this in?

File: src/app/build/build.component.ts [hunk 4]
```diff
@@ -1,15 +1,43 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, HostBinding } from '@angular/core';
+import { Router, ActivatedRoute, ParamMap } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/switchMap';
+
+import { MomentModule } from 'angular2-moment';
+
+import { BuildService } from '../services/build/build.service';
+// import { JobService } from '../job.service';
```
Comment (radu-matei) [inline src/app/build/build.component.ts]: And here we need the job service to get the logs?

File: src/app/build/build.component.ts [hunk 5]
```diff
@@ -1,15 +1,43 @@
...
+// import { ProjectService } from '../project.service';
 
 @Component({
   selector: 'app-build',
   templateUrl: './build.component.html',
   styleUrls: ['./build.component.scss']
 })
+
 export class BuildComponent implements OnInit {
+  build;
```
Comment (radu-matei) [inline src/app/build/build.component.ts]: Can we have this strongly-typed as well?

File: src/app/build/build.component.html [hunk 6]
```diff
@@ -6,7 +6,7 @@
 
         <div class="build-header">
           <h1>
-            <a href="../kashti/#!/project/brigade-830c16d4aaf6f5490937ad719afd8490a5bcbef064d397411043ac" title="Back to Build list">
+            <a routerLink="/projects/{{build.project_id}}">
```
Comment (flynnduism) [inline src/app/build/build.component.html]: Awesome, thanks for wiring that up

File: src/app/build/build.component.ts (hunk 4 above)
Comment (seanknox) [reply src/app/build/build.component.ts]: Nope! Removed.

**Comment by seanknox** [comment]
@radu-matei addressed all of your feedback, thanks!

**Comment by seanknox** [comment]
@flynnduism mergin'!

**Pull request merged**